### PR TITLE
(#48) Begin work of removing stale computers from CCM

### PR DESCRIPTION
**Pull request by steviecoaster**
This commit lays the ground work for a mechanism to remove stale machines from the CCM database
While this is scaffolded out, design decisions may yet need to be made.

### CODE REVIEW

**Review by adilio — APPROVED** [review]
So far so good, LGTM!

File: src/Public/Remove-CCMStaleComputer.ps1 [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    
+    .NOTES
+    
+    #>
+    [cmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    Param(
+        [Alias('Id')]
+        [Parameter(Mandatory, ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [Int[]]
+        $ComputerId,
```
Comment (JPRuskin) [inline src/Public/Remove-CCMStaleComputer.ps1:29]:
```suggestion
        [Parameter(Mandatory, ParameterSetName = 'Name')]
        [string[]]
        $Name,
        
        [Alias('Id')]
        [Parameter(Mandatory, ValueFromPipeline, ValueFromPipelineByPropertyName, ParameterSetName = 'Id')]
        [Int[]]
        $ComputerId = (Get-CCMComputer -Computer $Name).Id,
```

I'm always a huge fan of this lazy pattern, though strictly speaking PSSA isn't.

Comment (steviecoaster) [reply src/Public/Remove-CCMStaleComputer.ps1]: Ha, this is basically how I've implemented this!

**Comment by gep13** [comment]
@steviecoaster is this still required based on the work that went out in CCM 0.10.0?

**Pull request closed without merging**